### ISSUE TITLE
plugin-less instantiates CSSPluginBase() when it is not running in a browser.

### ISSUE THREAD

**Issue by mikol-styra** [systemjs/plugin-less#6]
I am trying (unsuccessfully) to use plugin-less in conjunction with server-side rendering. The root cause of the problem is `CSSPluginBase.prototype.instantiate()`, which attempts to inject a style element into the DOM without `document` being defined:

``` javascript
CSSPluginBase.prototype.instantiate = function(load) {
  if (this.builder)
    return;

  var style = document.createElement('style');
  style.type = 'text/css';
  style.innerHTML = load.metadata.style;
  document.head.appendChild(style);
};

module.exports = CSSPluginBase;
```

I am not sure what the correct fix is in this case. There may be two. First, explicitly checking if `document` is defined in `CSSPluginBase.prototype.instantiate()` seems appropriate and works for me in practice:

``` javascript
  if (this.builder || typeof document === 'undefined')
    return;
```

Second, `less.js` may require a fix so that the `CSSPluginBase()` constructor is not even called unless plugin-less is running in a browser. plugin-css explicitly checks if it is running in a browser, for example, and does not have an issue with server-side rendering. Neither does mobilexag/plugin-sass.

**Comment by guybedford** [comment]
It should be fine to entirely short-circuit the CSS loading when running in a non-browser environment, with a check exactly like that.

If you think that will do it, please do send a PR. Or do we need to do something here as well?

**Comment by mikol-styra** [comment]
Will be fixed by systemjs/plugin-css/pull/113. After thinking about this for a bit, I like the fact that plugins built on top of `CSSPluginBase` are not required to worry about the browser injection issue.
